### PR TITLE
fix: Fix some compatibility issues related to older versions of K8s

### PR DESCRIPTION
**Pull request by CH3CHO**
### Ⅰ. Describe what this PR did

Fix some compatibility issues related to older versions of K8s
1. Support v1beta1/CustomResourceDefinitions used in K8s < 1.16.0
2. No to list v1beta1/IngressClass since there is never such a resource existed
3. No to monitor v1/MutatingWebhookConfiguration since it's in fact not needed

### CODE REVIEW

File: istio/1.12/patches/istio/20230603-unwatch-webhook.patch [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ 		mu:       sync.Mutex{},
+ 	}
+-	p.webhookInformer = client.KubeInformer().Admissionregistration().V1().MutatingWebhookConfigurations().Informer()
+-	p.webhookInformer.AddEventHandler(p.makeHandler())
++
++	// Commented by ingress
++	// Higress doesn't need to watch MutatingWebhookConfigurations, and it will break compatibility with K8s < v1.16.0.
++	// p.webhookInformer = client.KubeInformer().Admissionregistration().V1().MutatingWebhookConfigurations().Informer()
++	// p.webhookInformer.AddEventHandler(p.makeHandler())
++	// End commented by ingress
```
Comment (SpecialYang) [inline istio/1.12/patches/istio/20230603-unwatch-webhook.patch:15]: Istiod has a env INJECT_ENABLED which determines the inject behavior.

File: pkg/ingress/kube/ingress/status.go [hunk 2]
```diff
@@ -48,11 +48,12 @@ type statusSyncer struct {
...
-		controller:         controller,
-		watchedNamespace:   namespace,
-		ingressLister:      client.KubeInformer().Networking().V1beta1().Ingresses().Lister(),
-		ingressClassLister: client.KubeInformer().Networking().V1beta1().IngressClasses().Lister(),
+		client:           client,
+		controller:       controller,
+		watchedNamespace: namespace,
+		ingressLister:    client.KubeInformer().Networking().V1beta1().Ingresses().Lister(),
+		// There is no IngressClass resource in v1beta1
+		ingressClassLister: nil,
```
[the remark this answers is not in the document]
Comment (CH3CHO) [reply pkg/ingress/kube/ingress/status.go:56]: So should we delete it in status.go for v1 as well?

[the remark this answers is not in the document]
Comment (SpecialYang) [reply pkg/ingress/kube/ingress/status.go:56]: Yes.

File: istio/1.12/patches/istio/20230603-unwatch-webhook.patch (hunk 1 above)
Comment (SpecialYang) [reply istio/1.12/patches/istio/20230603-unwatch-webhook.patch:15]: https://github.com/alibaba/higress/blob/80d6ecfddb0c349dfc453787d0b89d253530dba4/helm/core/templates/controller-deployment.yaml#L75-L77

Add env INJECT_ENABLED: false here to disable this webhook.

**Review by johnlanni — CHANGES_REQUESTED** [review]

File: istio/1.12/patches/istio/20230603-unwatch-webhook.patch (hunk 1 above)
Comment (CH3CHO) [reply istio/1.12/patches/istio/20230603-unwatch-webhook.patch]: Done. Please check it out.

**Review by johnlanni — APPROVED** [review]

**Pull request merged**